python/pyarrow/parquet.py

```diff
@@ -849,6 +857,31 @@ def open_file(path, meta=None):
                                    common_metadata=self.common_metadata)
         return open_file
 
+    def _filter(self, filters):
+        def filter_accepts_partition(part_key, filter, level):
+            p_column, p_value_index = part_key
+            f_column, op, f_value = filter
+            if p_column != f_column:
+                return True
+
+            f_value_index = self.partitions.get_index(level, p_column,
+                                                      str(f_value))
+            if op == "=":
```

Comment:
Clarified offline with @jneuff that we will refactor `self.partitions.get_index` a bit to support more operation than the two above.